objects/obj_drop_select/Draw_0.gml

```diff
@@ -507,13 +507,22 @@ if (scr_hit(xx+954,yy+556,xx+1043,yy+579)=true){
             }
         }
         
+        // Iterates through all selected "ships" (max 30), including the planet (Local on the drop menu), 
+        // and fills the battle roster with any marines found.
         var i;i=-1;ships_selected=0;
         repeat(31){
             i+=1;if (ship_all[i]!=0) then scr_battle_roster(ship[i],ship_ide[i],false);
         }
-        if (ship_all[500]=1) and (attack=1) then scr_battle_roster(p_target.name,obj_controller.selecting_planet,true);
-        
         
+        // K-INI
+        // Trying to solve marines duplicating when selecting "All ships" (ship_all[500]=1) on the drop menu.
```

Comment:
if this solves the issue, then please remember to delete the commented out code

Reply:
Will do!